CONTRIBUTING.rst

```diff
@@ -106,12 +106,65 @@ Before you submit a pull request, check that it meets these guidelines:
    https://travis-ci.com/pyfar/pyfar/pull_requests
    and make sure that the tests pass for all supported Python versions.
 
-Tips
-----
 
-To run a subset of tests::
+Testing Guidelines
+-----------------------
+Test-Driven-Development (TDD) is the fundamental technique followed in developing the pyfar software packages to ensure a good software development practice. In principle, it is based on `three steps <https://martinfowler.com/bliki/TestDrivenDevelopment.html>`_
+
+- Write a test for the next bit of functionality you want to add.
+- Write the functional code until the test passes.
+- Refactor both new and old code to make it well structured.
```

Comment:
For the sake of keeping this short, we could also simply write:

Pyfar uses test-driven development `three steps <https://martinfowler.com/bliki/TestDrivenDevelopment.html>` and continuous integration `some link` to test and monitor the code. In the following...